rampwf/utils/testing.py

```diff
@@ -234,3 +242,91 @@ def blend_submissions(submissions, ramp_kit_dir='.', ramp_data_dir='.',
         save_output=save_output,
         score_table_title='Foldwise best bagged scores',
         score_f_name_prefix='combined')
+
+
+
+def _make_random_parameters_space(params_distribution, seed=None):
+    """Make a random set of hyper parameters sampled from `params_distribution`.
+
+    Parameters:
+    -----------
+    params_distribution: dict
+       a dictionay whose keys are hyperparameters names and values are iterables
+       of possible values for each parameter.
+    seed:
+       random state value used for shuffling. Default: None.
+
+    Returns:
+    --------
+    params_space: generator
+        generator of all random combinations of hyper parameters. Each element
+        will be a dictionary of parameters names and values.
+    """
+    param_names, param_values = list(zip(*params_distribution.items()))
+    space = list(itertools.product(*param_values))
+    random.seed(seed)
+    random.shuffle(space)
+    for values in space:
+        yield dict(zip(param_names, values))
+
+
+def _configure_submission(params, problem, submission_path, outdir):
+    """Configure submission with a set of [hyper-]parameters.
+    """
+    filename = os.path.join(submission_path, problem.workflow.element_names[1] + ".py")
+    # update hyperparameters section of estimator module
+    with open(filename) as fp:
+        text = fp.read()
+        text = HYPERPARAMS_REPL_REGEX.sub(HYPERPARAMS_REPL_PATTERN.format(params),
+                                          text)
+    with open(filename, "w") as fp:
+        fp.write(text)
+
+
+def _eval_submission_with_params(params, problem, submission_path, X_train, y_train):
+    print("Run with hyperparams:", params)
+    with tempfile.TemporaryDirectory() as dirname:
+        submission_name = os.path.split(submission_path.rstrip(os.path.sep))[1]
+        temp_submission_path = os.path.join(dirname, submission_name)
+        shutil.copytree(submission_path, temp_submission_path)
+
+        _configure_submission(params, problem, temp_submission_path, dirname)
+
+        df_scores_list = []
+        predictions_valid_list = []
+
+        for fold_i, fold in enumerate(problem.get_cv(X_train, y_train)):
```

Comment:
The hyperopt engine should have a granularity of running `params` on each cv fold separately, ie, being able to decide how many folds it needs. On a very bad combination we may need to run only a single fold, to save time. On the single fold results we should then store: `fold_i`, `n_train`, `n_test`, `score`, and even `train_time` and `test_time` (or even `RAM` if we have it). So the hyperopt algo has all the information to smartly choose the next params and `fold_i`.